packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions/test/integ.state-machine-and-activity-cmk.ts

```diff
@@ -0,0 +1,80 @@
+import * as cdk from 'aws-cdk-lib';
+import * as sfn from 'aws-cdk-lib/aws-stepfunctions';
+import * as task from 'aws-cdk-lib/aws-stepfunctions-tasks';
+import * as kms from 'aws-cdk-lib/aws-kms';
+import { ExpectedResult, IntegTest } from '@aws-cdk/integ-tests-alpha';
+
+class KMSStateMachine extends cdk.Stack {
+  readonly stateMachine: sfn.StateMachine;
+  readonly activity: sfn.Activity;
+  readonly kmsKey: kms.Key;
+
+  constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    this.kmsKey = new kms.Key(this, 'Key used for encryption', {
```

Comment:
Instead of using the same KMS key to encrypt the state machine and activity, it's probably best to use two separate ones. That way we can verify policy generation for the state machine and activity Task independently.

Reply:
Ok will update.